src/launchpad/parsers/android/dex/dex_mapping.py

```diff
@@ -0,0 +1,242 @@
+import re
+
+from dataclasses import dataclass
+
+from launchpad.parsers.android.dex.android_code_utils import AndroidCodeUtils
+
+
+@dataclass
+class DexMappingClass:
+    deobfuscated_fqn: str
+    deobfuscated_signature: str
+    obfuscated_fqn: str
+    file_name: str | None = None
+    # Start line within file. This may not be present even if fileName is present
+    start_line: int | None = None
+    # Method and field mappings for this class
+    methods: dict[str, str] | None = None  # obfuscated_name -> deobfuscated_name
+    fields: dict[str, str] | None = None  # obfuscated_name -> deobfuscated_name
+
+
+class DexMapping:
+    def __init__(self, bytes: bytes) -> None:
+        self._classes: dict[str, DexMappingClass] = {}
+
+        classes: list[DexMappingClass] = []
+        current_class: DexMappingClass | None = None
+        pending_file_name: str | None = None
+
+        lines = bytes.decode("utf-8").splitlines()
+
+        for line in lines:
+            if line.startswith("#"):
+                pending_file_name = self._parse_comment(line, pending_file_name)
+            elif line.startswith(" "):
+                trimmed = line.strip()
+                if trimmed.startswith("#"):
+                    pending_file_name = self._parse_comment(line, pending_file_name)
+                else:
+                    current_class = self._parse_method_or_member(line, current_class)
+            elif line.endswith(":"):
+                current_class, pending_file_name = self._parse_class(line, current_class, pending_file_name, classes)
```

Comment:
Is it worth having an else clause here or are we only parsing a subset of the file?